### PR TITLE
Exam selection now only displays channels that contain exercises

### PR DESCRIPTION
**Pull request by alanchenz**
### Summary

During exam creation, the drop down menu would have channels that didn't have exercises in them. This would be potentially be confusing and inconvenient for coaches who are trying to create exams. 

Now, only exams that actually have exercises are shown in the selection menu. This involved adding another feature in the content api.py file, which looks into each channel and checks if it has at least one exercise.

### Reviewer guidance

This can be tested by having multiple imported channels, some with exercises and some without exercise. Then, go to the coaching tab and create an exam. The exam selection menu should only show the channels that have exercises.

### References

This fixes issue #2770. 

### Contributor Checklist

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] Contributor has fully tested the PR manually
- [x] Screenshots of any front-end changes are in the PR description
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')

### Reviewer Checklist

- [ ] Automated test coverage is satisfactory
- [ ] Reviewer has fully tested the PR manually
- [ ] PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- [ ] External dependencies files were updated (`yarn` and `pip`)
- [ ] Documentation is updated
- [ ] Link to diff of internal dependency change is included
- [ ] CHANGELOG.rst is updated for high-level changes
- [ ] Contributor is in AUTHORS.rst

### CODE REVIEW

File: kolibri/core/assets/src/conditionalPromise.js [hunk 1]
```diff
@@ -14,12 +14,12 @@ export default class ConditionalPromise {
   }
 
   catch(...args) {
-    this._promise.catch(...args);
+    this._promise = this._promise.catch(...args);
```
Comment (indirectlylit) [inline kolibri/core/assets/src/conditionalPromise.js]: commit 57ac9b7f4f40215d4cfd61ba5d4ce84fd6c37055 should not be part of this PR.

to make a new PR with only this work, please:

* make a new branch
* `reset --hard` to the 0.8.x branch
* cherry-pick the three commits in this branch
* submit a new PR

thanks!

**Comment by indirectlylit** [comment]
also, looks like we're getting some linting errors:

```
pythonlint runtests: commands[0] | flake8 kolibri
kolibri/content/api.py:14:1: F401 'kolibri.content.models.ChannelMetadata' imported but unused
kolibri/content/api.py:21:1: F811 redefinition of unused 'BooleanFilter' from line 12
kolibri/content/api.py:22:1: F811 redefinition of unused 'CharFilter' from line 12
kolibri/content/api.py:23:1: F811 redefinition of unused 'ChoiceFilter' from line 12
kolibri/content/api.py:24:1: F811 redefinition of unused 'DjangoFilterBackend' from line 12
kolibri/content/api.py:25:1: F811 redefinition of unused 'FilterSet' from line 12
kolibri/content/api.py:26:1: F811 redefinition of unused 'content_kinds' from line 18
kolibri/content/api.py:27:1: F811 redefinition of unused 'languages' from line 18
kolibri/content/api.py:28:1: F811 redefinition of unused 'mixins' from line 19
kolibri/content/api.py:29:1: F811 redefinition of unused 'pagination' from line 19
kolibri/content/api.py:30:1: F811 redefinition of unused 'viewsets' from line 19
kolibri/content/api.py:31:1: F811 redefinition of unused 'detail_route' from line 20
kolibri/content/api.py:32:1: F811 redefinition of unused 'list_route' from line 20
kolibri/content/api.py:37:1: F811 redefinition of unused 'models' from line 13
kolibri/content/api.py:38:1: F811 redefinition of unused 'serializers' from line 13
kolibri/content/api.py:39:1: F811 redefinition of unused 'CanManageContent' from line 15
kolibri/content/api.py:40:1: F811 redefinition of unused 'get_channel_lookup_url' from line 16
kolibri/content/api.py:41:1: F811 redefinition of unused 'ContentSessionLog' from line 17
kolibri/content/api.py:42:1: F811 redefinition of unused 'ContentSummaryLog' from line 17
kolibri/content/api.py:51:33: E231 missing whitespace after ','
kolibri/content/api.py:51:42: E231 missing whitespace after ','
kolibri/content/api.py:51:47: E231 missing whitespace after ','
kolibri/content/api.py:64:46: E231 missing whitespace after ','
ERROR: InvocationError: '/home/travis/build/learningequality/kolibri/.tox/pythonlint/bin/flake8 kolibri'
___________________________________ summary ____________________________________
ERROR:   pythonlint: commands failed
```

**Comment by alanchenz** [comment]
Superseded by pull request #3325.
